webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala

```diff
@@ -1154,27 +1176,35 @@ object ExecutionBatch extends Loggable {
     // bugs like #7758. A pattern specificity, in our case, can somehow be told from
     // the lenght of the pattern when \Q\E.* are removed.
     //
-    val values = expectedComponent.groupedComponentValues.toList.map { case(v, u) =>
+    val values = expectedComponent.groupedComponentValues.toList.map { case (v, u) =>
       val isVar = matchCFEngineVars.pattern.matcher(v).matches()
-      val pattern = replaceCFEngineVars(v)
-      val specificity = pattern.toString.replaceAll("""\\Q""", "").replaceAll("""\\E""", "").replaceAll("""\.\*""", "").size
+      val pattern = if (isVar) { // If this is not a var, there isn't anything to replace.
+        Some(replaceCFEngineVars(v))
+      } else {
+        None
+      }
+      //If this is not a variable, we use the variable itself
+      val specificity = pattern.map(_.toString.replaceAll("""\\Q""", "").replaceAll("""\\E""", "").replaceAll("""\.\*""", "")).getOrElse("v").size
```

Comment:
getOrElse("v") instead of getOrElse(v)
:facepalm:

Reply:
Type was ok at least :) it works for a sub part of all cases (those were the value is v) so it was not all wrong